src/poliastro/twobody/propagation.py

```diff
@@ -189,6 +192,134 @@ def _kepler(k, r0, v0, tof, *, numiter):
     return r, v
 
 
+def _gauss(k, r, v, tof, rtol=1e-10, numiter=150, **kwargs):
+    """ Propagates orbit the specified tof. """
+
+    # Work with international system
+
+    k = k.to(u.m ** 3 / u.s ** 2).value
+    dot_rv = np.dot(r.to(u.m).value, v.to(u.m / u.s).value)
+    r0 = norm(r).to(u.m).value
+    v0 = norm(v).to(u.m / u.s).value
+    tof = tof.to(u.s).value
+
+    # Computing main variables of algorithm
+    beta0 = np.sqrt(k / r0 ** 3)
+    delta0 = r0 * v0 ** 2 / k
+
+    phi0 = dot_rv / np.sqrt(k * r0)
+    phi0_prime = 0.5 * phi0
+    phi0_2prime = 0.5 * phi0_prime
+
+    T_prime = 3 * beta0 * tof
+    A_m = 3 / 10
+
+    gamma0_prime = 0.5 - 0.25 * delta0
+    gamma0_2prime = delta0 - 1
+
+    chi0_prime = 9 / 40 * delta0 - 1 / 5
+
+    # Algorithm starts iteration
+    A_old = 0
+    n = 0
+    while n < numiter:
+        c_coeff = [
+            8 / 175,
+            8 / 525,
+            1896 / 336875,
+            28744 / 13138125,
+            26248 / 29859375,
+            134584 / 372246875,
+            129802986344 / 857740555546875,
+            55082676858 / 857740555546875,
+            687061097149992 / 24934041427216796875,
+            8033038585237352 / 673219118534853515625,
+            2892031498456202296 / 555405772791254150390625,
+            4093458329892811912 / 1789640823438485595703125,
+            10507274811793509548037032 / 10398126371321703046014404296875,
+            25259289756593760965336 / 56299726292037542523193359375,
+        ]
+
+        C = 0
+        for idx, c_i in enumerate(c_coeff):
+            C += c_i * A_old ** (idx + 2)
+
+        alpha = 1 + 1 / 5 * A_old + C
+        alpha_prime = 1 - 3 / 5 * A_old
+
+        psi = alpha * T_prime
+        eta = alpha_prime * alpha_prime * psi / (alpha - A_old)
+
+        epsilon = 1 + eta * phi0_prime
+        b = np.abs(epsilon + eta * (phi0_2prime + chi0_prime * psi))
+
+        def f(x, epsilon, b):
+            """ This is eq. 5.83 in Battin. """
+            return x ** 3 - 3 * epsilon * x - 2 * b
+
+        # Solving for the roots
+        x = root(f, b, args=(epsilon, b)).x[0]
```

Comment:
Is there a chance we have the derivative of the function? I think the answer is yes, so I would use `newton` instead. Even if we don't provide the derivative, I would not use `root`, because it's too general purpose and impossible to accelerate with numba (unless we explore https://github.com/poliastro/poliastro/issues/486)